core/runtime/src/main/java/io/quarkus/runtime/ShutdownContext.java

```diff
@@ -12,10 +14,21 @@
  */
 public interface ShutdownContext {
 
-    void addShutdownTask(Runnable runnable);
+    int DEFAULT_PRIORITY = Interceptor.Priority.LIBRARY_AFTER;
```

Comment:
I wonder if you need your own `Priority` class with its own set of constants... Because here the "library after" wording is kind of confusing: applying an interceptor "after" the application has quite different implications than what we're doing here. A dedicated class with constants would at least be able to only provide a few constants that make sense, and document them.